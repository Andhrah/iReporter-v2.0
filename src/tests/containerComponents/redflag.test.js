import React from 'react';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../store';
import ConnectedRedFlag, { RedFlag } from '../../components/container/RedFlag';

const props = {
  getSingleRedflag: jest.fn(n => n * n),
  match: { params: { id: 1 } },
  singleRedFlag: { data: ['first value', 'second value'], loading: true },
};

describe('RedFlag test suite', () => {
  it('renders correctly', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ConnectedRedFlag {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('handles component did mount correctly', () => {
    const wrapper = shallow(
      <RedFlag {...props} />
    );
    wrapper.instance().componentDidMount();
    expect(wrapper.instance().props.getSingleRedflag).toHaveBeenCalled();
  });
});
