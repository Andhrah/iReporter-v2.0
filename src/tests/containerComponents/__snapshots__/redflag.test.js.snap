// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedFlag test suite renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "1lpkn1",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "1lpkn1",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ConnectFunction
        getSingleRedflag={[MockFunction]}
        match={
          Object {
            "params": Object {
              "id": 1,
            },
          }
        }
        singleRedFlag={
          Object {
            "data": Array [
              "first value",
              "second value",
            ],
            "loading": true,
          }
        }
      >
        <RedFlag
          getSingleRedflag={[Function]}
          match={
            Object {
              "params": Object {
                "id": 1,
              },
            }
          }
          singleRedFlag={
            Object {
              "data": Array [
                Object {
                  "comment": "",
                  "corruption_date": "",
                  "corruption_methods": Array [],
                  "created_by": null,
                  "created_on": "",
                  "entity_involved": Array [],
                  "id": null,
                  "images": Array [],
                  "location": "",
                  "names_involved": Array [],
                  "status": "",
                  "videos": Array [],
                },
              ],
              "loading": true,
            }
          }
        />
      </ConnectFunction>
    </Router>
  </MemoryRouter>
</Provider>
`;
